src/sonic_ax_impl/mibs/vendor/cisco/ciscoSwitchQosMIB.py

```diff
@@ -90,7 +96,9 @@ def update_data(self):
         """
         for queue_key, sai_id in self.port_queues_map.items():
             queue_stat_name = mibs.queue_table(sai_id)
-            queue_stat = Namespace.dbs_get_all(self.db_conn, mibs.COUNTERS_DB, queue_stat_name, blocking=False)
+            if_index, _ = queue_key.split(':')
+            queue_stat_idx = mibs.queue_key(if_index, queue_stat_name)
+            queue_stat = self.queue_stat_map.get(queue_stat_idx, {})
```

Comment:
can queue_stat  be None ? Do we have to do this check

Reply:
> can queue_stat be None ? Do we have to do this check

We have a check in the line below to check if queue_stat is not None.

Reply:
my point was can it be ever None ? do we need to put if condition ?

Reply:
My mistake, none check is required the same as in: https://github.com/SuvarnaMeenakshi/sonic-snmpagent/blob/counters_fix/src/sonic_ax_impl/mibs/__init__.py#L358.
I have fixed the code to do a get_all to get the updated stats. 
The previous code was providing the correct results as each time reinit_data was invoked, the stats was getting updated.
But update data is called more frequently, and update_data also should do a get_all each time, after which the None check will be required.